src/libs/conduit/conduit_utils.cpp

```diff
@@ -72,6 +73,84 @@ namespace conduit
 namespace utils
 {
 
+void *
+default_alloc_handler(size_t items, size_t item_size)
+{
+  return calloc(items, item_size);
+}
+
+void
+default_free_handler(void *data_ptr)
+{
+  free(data_ptr);
+}
+
+void default_memset_handler(void * ptr, int value, size_t num )
+{
+  memset(ptr,value,num);
+}
+
+void default_memcpy_handler(void * destination, const void * source, size_t num)
+{
+  memcpy(destination,source,num);
+}
+
+static std::map<int32,void*(*)(size_t, size_t)> allocator_map
+  = {{0, &default_alloc_handler}};
+
+static std::map<int32,void(*)(void*)> free_map
+  = {{0, default_free_handler}};
+
+static std::map<int32,void(*)(void*,const void *,size_t)> memcpy_map
+  = {{0, default_memcpy_handler}};
+
+static std::map<int32,void(*)(void*,int,size_t)> memset_map
+  = {{0, default_memset_handler}};
+
+
+
+int32 register_mem_handler( void*(*allocate) (size_t, size_t),
+                            void(*free)(void *),
+                            void(*copy)(void*,const void *,size_t),
+                            void(*memset)(void*,int,size_t))
+{
+  static int32 allocator_id = 1;
+  allocator_map[allocator_id] = allocate;
+  free_map[allocator_id] = free;
+  memcpy_map[allocator_id] = copy;
+  memset_map[allocator_id] = memset;
```

Comment:
I think I need to change these names. I can totally see a compiler using the other free, not the provided one.

Reply:
agreed, can we also name this `register_memory_handler`

Reply:
i updated these names